examples/simple/main.cc

```diff
@@ -0,0 +1,36 @@
+#include "opentelemetry/sdk/trace/simple_processor.h"
+#include "opentelemetry/sdk/trace/tracer_provider.h"
+#include "opentelemetry/trace/provider.h"
+
+// Using an exporter that simply dumps span data to stdout.
+#include "stdout_exporter.h"
+
+#include "foo_library/foo_library.h"
+
+namespace
+{
+void initTracer()
+{
+  // Specify the exporter: StdoutExporter in this case simply passes all output from the span
+  // processor to stdout
+  auto exporter = std::unique_ptr<sdktrace::SpanExporter>(new StdoutExporter);
+  // Specify the Span Processor: SimpleSpanProcessor forwards all completed spans directly to the
+  // exporter
+  auto processor = std::shared_ptr<sdktrace::SpanProcessor>(new sdktrace::SimpleSpanProcessor(
+      std::move(exporter)));  // Create a Span Processor given the StdoutExporter defined earlier
+  // Initialize a trace provider with the Span Processor instance defined above
+  auto provider = nostd::shared_ptr<trace::TracerProvider>(new sdktrace::TracerProvider(processor));
```

Comment:
I'd be more sparse with inline comments and only use them to hint at things that cannot be inferred from the code itself. Otherwise they tend to make code harder to read instead of making it easier. There's no information in
```
// Initialize a trace provider with the Span Processor instance defined above
```
that cannot be inferred from
```
auto provider = nostd::shared_ptr<trace::TracerProvider>(new sdktrace::TracerProvider(processor));
```

Reply:
Remind me on one thing that I learned long time ago - make the code talk by itself.